src/Cli/dotnet/Commands/Tool/Update/ToolUpdateCommandParser.cs

```diff
@@ -33,6 +28,7 @@ private static CliCommand ConstructCommand()
         CliCommand command = new("update", LocalizableStrings.CommandDescription);
 
         command.Arguments.Add(PackageIdArgument);
+        PackageIdArgument.Arity = ArgumentArity.ZeroOrOne;
```

Comment:
There probably is a better way to do this

Reply:
If you find yourself wanting to use the same 'core' of an argument/option, but changing it _slightly_ - make the default definition of it a method instead, and call that method to get a new copy of the same 'logical' argument/option each time/way you need to customize it. For example, [`--interactive`](https://github.com/dotnet/sdk/blob/063e87fab287fc92b2a38fee326ae7d27c1bf766/src/Cli/dotnet/CommonOptions.cs#L121-L139). We have a common, shared _core_ in a method, and we created one sharable form for folks that needed it to be forwarded to MSBuild. But there are a few other ways of using it (i.e. NuGet) so in the other places that use it they call the method to get a new copy and customize it to their use case (arity, forwarding, etc.).

Reply:
Oh that is a great idea, thanks!